File: 0179-largest-number/0179-largest-number.cpp
class Solution {
public:
    static bool cmp(string a, string b){
        return a+b > b+a;
    }

    string largestNumber(vector<int>& nums) {
        string res = "";
        
        vector<string> numbers;
        
        for (int iter: nums) {
            numbers.push_back(to_string(iter));
        }
        
        sort(numbers.begin(), numbers.end(), cmp);
        if (numbers[0] == "0") {
            return "0";
        }
        
        for (string n: numbers) {
            res += n;
        }
        return res;
    }
};

File: 0179-largest-number/0179-largest-number.py
def large_num_key(x, y):
    return -1 if x + y > y + x else 1


class Solution:
    def largestNumber(self, nums: List[int]) -> str:
        nums = [str(x) for x in nums]
        largest_nums = ''.join(sorted(nums, key=cmp_to_key(large_num_key)))
        return '0' if largest_nums[0] == '0' else largest_nums